cluster/service.go

```diff
@@ -167,6 +177,23 @@ func (s *Service) serve() error {
 	}
 }
 
+func (s *Service) checkCommandPerm(c *Command, perm string) bool {
+	if s.credentialStore == nil {
+		return true
+	}
+
+	var username string
```

Comment:
You could remove a block in your `if` statement by simply setting username and password to "" to begin with. I.e.
```Go
username := ""
password := ""
if c.Credentials != nil {
                 username = c.Credentials.GetUsername()
		password = c.Credentials.GetPassword()
}
```